lib/org/chromium/webidl/IDLFragmentExtractor.js

```diff
@@ -21,96 +22,179 @@ foam.CLASS({
+        // <div> tag while <p> is on top of the stack, we pop <p>.
+        if (top && top.nodeName === 'p' && tag.nodeName === 'div') {
+          this.tagStack_.pop();
+        }
+
+        if (this.tagStack_.excludedTop === null && tag.nodeName === 'pre' && isIDL) {
+          // Found a <pre class="idl.*">.
+          // Ignore tags and only extract text now.
+          this.tagMatching_ = false;
+        }
+        this.tagStack_.push(tag);
+      },
+    },
+    {
+      name: 'extractPreContents',
+      documentation: 'Action performed when extracting contents inside pre tag.',
+      code: function(i, ignore) {
+        var content = '';   // Used to group together related content.
+        var incomplete = true;
+
+        while(incomplete) {
+          var item = this.tags_[++i];
+          var isTag = this.lexer_.Tag.isInstance(item);
+
+          // FUTURE: Handle nested pre tags.
+          // As of this writing, there has not been any IDL fragments
+          // that has been found nested within other pre tags.
+          if (isTag && item.nodeName === 'pre') {
+            this.tagStack_.pop();
+            incomplete = false;
+
+            if (!ignore) {
+              this.idlFragments.push(this.lexer_.lib.unescapeString(content));
             }
-          } else {
-            // item is text fragments, so we append it.
+          } else if (!isTag) {
+            // Ignoring all tags. Only extracting text content within pre tags.
             content += item;
           }
-        } else if (isTag) {
-          // Encountered a tag and we are tag matching.
-          // Perform appropriate action based on class attribute and tagType.
-          var top = tagStack[tagStack.length - 1];
-          var classes = extractAttr(item, 'class');
-          var isIDL = classes && classes.includes('idl');
-          var isExcluded = function(cls) {
-            return cls && (cls.includes('example') || cls.includes('note'));
-          };
-
-          if (item.type.name === OPEN) {
-            if (isExcluded(classes)) {
-              // Entering the body of an excluded tag.
-              excludeStack.push(item);
-            } else if (excludeStack.length === 0
-                && item.nodeName === 'pre' && isIDL) {
-              // Found a <pre class="idl.*">.
-              // Ignore tags and only extract text now.
-              tagMatching = false;
+        }
+        return i;
+      },
+    },
+    {
+      name: 'extract',
+      documentation: 'Extracts all IDL Fragments from file property.',
+      code: function() {
+        var OPEN = this.TagType.OPEN.name;
+        var CLOSE = this.TagType.CLOSE.name;
+        var tagStack = this.tagStack_;
+        var ignoreNextPre = false;    // Set to ignore content of next pre tag.
+
+        for (var i = 0; i < this.tags_.length; ++i) {
+          var item = this.tags_[i];
+          var isTag = this.lexer_.Tag.isInstance(item);
+
+          if (isTag) {
```

Comment:
Sorry, I'm not too sure what you mean by "save an indent". 

The item will be a text fragment if it is not a tag at this point. Since we are not extracting this text content (i.e. it is not in a `pre` block), they are simply discarded.